packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1604,6 +1606,24 @@ class OverlayPortalController {
   /// rebuilt.
   void toggle() => isShowing ? hide() : show();
 
+  final List<VoidCallback> _listeners = <VoidCallback>[];
+
+  void _notifyListeners() {
+    for (final VoidCallback listener in _listeners) {
```

Comment:
If you `extend ChangeNotifier` it gives you an implementation that would allow you to add/remove listeners in the listener callback. With this implementation you'll get a concurrent modification exception if you try to do that.